File: engine/engine-model/src/main/scala/net/server/channel/packet/cash/use/UseAvatarMegaphone.scala
package net.server.channel.packet.cash.use

class UseAvatarMegaphone(private var _position: Short, private var _itemId: Int, private var _messages: Array[String], private var _ear: Boolean) extends AbstractUseCashItemPacket(_position, _itemId) {
  def messages: Array[String] = _messages

  def ear: Boolean = _ear
}

File: engine/engine-model/src/main/java/net/server/channel/packet/cash/use/UseAvatarMegaphone.java
package net.server.channel.packet.cash.use;

public class UseAvatarMegaphone extends AbstractUseCashItemPacket {
   private final String[] messages;

   private final Boolean ear;

   public UseAvatarMegaphone(Short position, Integer itemId, String[] messages, Boolean ear) {
      super(position, itemId);
      this.messages = messages;
      this.ear = ear;
   }

   public String[] messages() {
      return messages;
   }

   public Boolean ear() {
      return ear;
   }
}
